packages/back-end/src/routers/organizations/organizations.controller.ts

```diff
@@ -1218,17 +1230,60 @@ export async function postApiKey(
 
   // Only require permissions if we are creating a new API key
   if (secret) {
-    req.checkPermissions("manageApiKeys");
+    if (type !== "user") {
+      // All access token types except `user` require the permission
+      req.checkPermissions("manageApiKeys");
+    }
   } else {
     req.checkPermissions("manageEnvironments", "", [environment]);
   }
 
-  const key = await createApiKey({
-    organization: org.id,
-    description: description || "",
-    environment: environment || "",
-    project: project || "",
-    secret: !!secret,
+  // Handle user personal access tokens
+  if (type === "user") {
+    if (!req.userId) {
```

Comment:
8ad7e5677e35551725491ccbc6f2eb0bddfc193d

Reply:
It looks like that's just an abstraction that uses `req.userId` but i made the change since it seems that's the preferred way to do it. Based on the name of the method, it didn't look like it would return the userId and only the organization.